internal/storage/postgres/migrations/20250507070019_add_plugin_policy_sync.sql

```diff
@@ -0,0 +1,19 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE plugin_policy_sync (
+    id UUID PRIMARY KEY,
+    policy_id UUID NOT NULL,
+    sync_type INT NOT NULL,
+    status INT NOT NULL,
```

Comment:
Personally, I am partial to using postgres enums to represent fixed-valueset columns

It can be used with pgx relatively painlessly: https://github.com/jackc/pgx/issues/338#issuecomment-333399112

Not a blocker for this PR - I will do a round of db refactoring next week, I can put it into that